### DISCUSSION TITLE
Performance difference between Linux FB and Skia on GTK3 (in Wayland/X11)?

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#12960] [Q&A] [answered]
I currently have my project/app running on RPi4 in framebuffer mode. It works fairly well, but I notice some UI performance issues, for example I have a virtual keyboard on the screen, about 30 buttons, it's dog slow on the RPi (runs fine on my PC in Skia). I haven't tried to use an X environment on the RPi, as I haven't found a good reference solution for it (I'm running my solution in a privileged docker container), so before I invest the time to get an X environment set up I wanted to ask if I can expect a good UI performance increase with Skia on GTK3? And if so, what should the appropriate X environment consist of, Wayland, X11, Weston, etc? What I've found online seems fairly outdated, so I wanted to ask first.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report.

Indeed, framebuffer is generally be slower, particularly on larger screens as the rendering of the UI is performed using Skia's software backend. Using GTK, OpenGL (or OpenGL ES) is enabled and makes for significantly faster rendering. Wayland or X11 can be used, as long as the OpenGL environment is setup.

When you're saying that what you found online is outdated, could you point to what you found so we can adjust it if possible?

**Reply by HakanL** [reply]
Ah, thanks for the insight. I wasn't clear, the outdated information wasn't from Uno, it was X11/Wayland-information in a docker container that I wasn't able to find any updated information about. To your knowledge, is Wayland a better option than X11? I'm looking to run my Uno application in 100% kiosk mode, never any other applications so I don't (think I) need a window manager (I'm a little rusty on X on Linux). Is there a tutorial for how to run Uno in Kiosk mode (no other apps) on RPi? If not, that would be great! :)
